client/src/app/appCtrl.js

```diff
@@ -46,36 +60,127 @@ angular.module('app').controller('appCtrl', ['$scope', '$http', '$timeout', 'app
     slidingWindow = false;
     $scope.$broadcast("fileUploadChange");
     $scope.view.loading = true;
-    // we do a quick test here to see if the server supports the Range header.
-    // If so, we try to stream. If not, we try to download.
-    $http.head($scope.view.filePath,{'headers' : {'Range' : 'bytes=0-32'}}).then(function(response){
-      if(response.status === 206) {
-        // now we check to see how big the file is
-        $http.head($scope.view.filePath).then(function(response){
-          var contentLength = response.headers('Content-Length');
-          if (contentLength > appConfig.MAX_FILE_SIZE) {
-            slidingWindow = true;
-            $scope.view.windowing.show = true;
-          }
-          streamRemoteFile($scope.view.filePath);
-        });
-      } else {
-        downloadFile($scope.view.filePath);
-      }
-    }, function() {
-      downloadFile($scope.view.filePath);
-    });
+    $scope.view.file.size = getFileSize($scope.view.filePath);
```

Comment:
This won't work, because the getFileSize function contains an asynchronous HTTP call. The lines below that depend on the output of the function will be run before the call has been returned. The proper way to do this is to put all of the logic that depends on the value of $scope.view.file.size inside of the then() function. This is because the $http.get() method returns a "promise", which in Angular, is a way of dealing with asynchronous code. There is more information here about how to work with promises: https://docs.angularjs.org/api/ng/service/$q#the-promise-api